### PR TITLE
Updated to 2.0 management lib

### PR DESCRIPTION
**Pull request by brnleehng**

### CODE REVIEW

File: CSharp/AccountManagement/AccountManagement.csproj [hunk 1]
```diff
@@ -45,9 +45,9 @@
...
       <HintPath>packages\Microsoft.Azure.Common.2.1.0\lib\net45\Microsoft.Azure.Common.NetFramework.dll</HintPath>
     </Reference>
-    <Reference Include="Microsoft.Azure.Management.Batch, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>packages\Microsoft.Azure.Management.Batch.1.5.0\lib\net40\Microsoft.Azure.Management.Batch.dll</HintPath>
+    <Reference Include="Microsoft.Azure.Management.Batch, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>packages\Microsoft.Azure.Management.Batch.2.0.0\lib\net45\Microsoft.Azure.Management.Batch.dll</HintPath>
+      <Private>True</Private>
     </Reference>
     <Reference Include="Microsoft.Azure.ResourceManager, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (jasper-schneider) [inline CSharp/AccountManagement/AccountManagement.csproj]: If you're feeling brave, this code could actually update to the new Resource package, since we only call the ResourceManagementClient methods ourselves.

**Comment by jasper-schneider** [comment]
This project is linked with an article: https://azure.microsoft.com/en-us/documentation/articles/batch-management-dotnet/

You may want to sync with Marsh to at least let him know that you're making these changes.

File: CSharp/AccountManagement/Program.cs [hunk 2]
```diff
@@ -1,334 +1,349 @@
...
+                // Regenerate primary account key
+                Console.WriteLine("Regenerating the primary key of account {0}...", account.Name);
+                BatchAccountRegenerateKeyResult newKeys = await batchManagementClient.Account.RegenerateKeyAsync(
+                    ResourceGroupName, account.Name,
+                    new BatchAccountRegenerateKeyParameters() { KeyName = AccountKeyType.Primary });
+                Console.WriteLine("  New primary key of account {0}: {1}", account.Name, newKeys.Primary);
+                Console.WriteLine("  Secondary key of account {0}:   {1}", account.Name, newKeys.Secondary);
+                Console.WriteLine();
+
+                // Print subscription quota information
```
Comment (jasper-schneider) [inline CSharp/AccountManagement/Program.cs]: This comment seems inaccurate.  You're printing the total number of accounts in the subscription here.

File: CSharp/AccountManagement/Program.cs [hunk 3]
```diff
@@ -1,334 +1,349 @@
...
+
+                // Delete account
+                Console.Write("Hit ENTER to delete account {0}: ", account.Name);
+                Console.ReadLine();
+                Console.WriteLine("Deleting account {0}...", account.Name);
+                try
+                {
+                    await batchManagementClient.Account.DeleteAsync(ResourceGroupName, account.Name);
+                }
+                catch (CloudException ex)
```
Comment (jasper-schneider) [inline CSharp/AccountManagement/Program.cs]: Hmm, I feel like we need a comment here.  Something like this?

"Account deletion is a long running operation. This .DeleteAsync() method will submit the account deletion request and poll for the status of the long running operation until the account is deleted. Currently, querying for the operation status after the account is deleted will return a 404 error, so we have to add this catch statement. This behavior will be fixed in a future service release."

Alternatively, since this is a sample, we have a little more leeway in how we approach the problem than in our SDKs.  We could simplify this check to just continue on all 404s.  If the account never existed, it's arguable that calling the Delete method shouldn't be a terminating error.  We should probably still include the comment to explain where the 404 comes from though, since that's unexpected.

if (ex.Response.HttpStatusCode != HttpStatusCode.NotFound)
{
    throw;
}

File: CSharp/AccountManagement/Program.cs [hunk 4]
```diff
@@ -1,334 +1,349 @@
...
-            }
-        }
-
-        private static async Task MainAsync()
-        {
-            // Obtain an access token using the "common" AAD resource. This allows the application
-            // to query AAD for information that lies outside the application's tenant (such as for
-            // querying subscription information in your Azure account).
-            AuthenticationContext authContext = new AuthenticationContext(AuthorityUri);
-            AuthenticationResult authResult = authContext.AcquireToken(ResourceUri,
```
Comment (Tatsinnit) [inline CSharp/AccountManagement/Program.cs]: 💡 - Idea/Note - There is a formatting issue that's why the changes shows +364 -339.
For anyone wondering about the exact changes in this pull review: check this link which takes out whitespaces, https://github.com/Azure/azure-batch-samples/pull/135/files?w=   
Thanks,

File: CSharp/AccountManagement/AccountManagement.csproj [hunk 5]
```diff
@@ -45,9 +45,9 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>packages\Microsoft.Azure.Common.2.1.0\lib\net45\Microsoft.Azure.Common.NetFramework.dll</HintPath>
     </Reference>
-    <Reference Include="Microsoft.Azure.Management.Batch, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
```
Comment (Tatsinnit) [inline CSharp/AccountManagement/AccountManagement.csproj]: 💡 Quick note here: Just wondering why the `specificversion` is removed? is it VS doing it for you without informing you : ) details as to how this works is detailed here: http://stackoverflow.com/questions/24022134/how-exactly-does-the-specific-version-property-of-an-assembly-reference-work-i

File: CSharp/AccountManagement/Program.cs [hunk 6]
```diff
@@ -1,334 +1,349 @@
...
+                Console.Write("Hit ENTER to delete account {0}: ", account.Name);
+                Console.ReadLine();
+                Console.WriteLine("Deleting account {0}...", account.Name);
+                try
+                {
+                    await batchManagementClient.Account.DeleteAsync(ResourceGroupName, account.Name);
+                }
+                catch (CloudException ex)
+                {
+                    if (!(ex.Request.Method == HttpMethod.Get && ex.Message.Contains("Long running operation failed with status 'NotFound'")))
```
Comment (Tatsinnit) [inline CSharp/AccountManagement/Program.cs]: 💡 a quick note: `.Contains` is case sensitive, in case you want to make it case insensitive, then worth adding `OrdinalIgnoreCase` https://msdn.microsoft.com/en-us/library/dy85x1sa(v=vs.110).aspx
